cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -619,3 +631,82 @@ func getDockerOptionsWithDefaults(options DockerProjectOptions) DockerProjectOpt
 
 	return options
 }
+
+func addDefaultDockerfileForJava(serviceConfig ServiceConfig) (string, error) {
+	// if Dockerfile not exists, provide a default one
+	if serviceConfig.Language == ServiceLanguageJava {
```

Comment:
`serviceConfig.Language == ServiceLanguageJava`, this check happened twice. How about doing like this:

```go
addDefaultDockerfileForJavaProject(serviceConfig) // Pass pointer

...


func addDefaultDockerfileForJava(serviceConfig ServiceConfig) (string, error) {
  if serviceConfig.Language == ServiceLanguageJava && serviceConfig.Docker.Path == "" {...}
}
```